server/express/src/controllers/movieController.ts

```diff
@@ -516,3 +522,806 @@ export async function findAndDeleteMovie(
+ * Demonstrates MongoDB $lookup aggregation to join collections.
+ */
+export async function getMoviesWithMostRecentComments(
+  req: Request,
+  res: Response
+): Promise<void> {
+  const moviesCollection = getCollection("movies");
+  const { limit = "10", movieId } = req.query;
+
+  const limitNum = Math.min(Math.max(parseInt(limit as string) || 10, 1), 50);
+
+  // Build aggregation pipeline
+  const pipeline: Document[] = [
+    // STAGE 1: Initial filter for data quality - filters bad data in
+    // the collection
+    {
+      $match: {
+        year: { $type: "number", $gte: 1800, $lte: 2030 },
+      },
+    },
+  ];
+
+  // Add movie ID filter if provided
+  if (movieId && typeof movieId === "string") {
+    if (!ObjectId.isValid(movieId)) {
+      res
+        .status(400)
+        .json(
+          createErrorResponse("Invalid movie ID format", "INVALID_OBJECT_ID")
+        );
+      return;
+    }
+    pipeline[0].$match._id = new ObjectId(movieId);
+  }
+
+  // Add remaining pipeline stages
+  pipeline.push(
+    // STAGE 2: Join with comments collection
+    {
+      $lookup: {
+        from: "comments",
+        localField: "_id",
+        foreignField: "movie_id",
+        as: "comments",
+      },
+    },
+    // STAGE 3: Filter movies with at least one comment
+    {
+      $match: {
+        comments: { $ne: [] },
+      },
+    },
+    // STAGE 4: Add computed fields
+    {
+      $addFields: {
+        recentComments: {
+          $slice: [
+            {
+              $sortArray: {
+                input: "$comments",
+                sortBy: { date: -1 },
+              },
+            },
+            limitNum,
+          ],
+        },
+        mostRecentCommentDate: {
+          $max: "$comments.date",
+        },
+      },
+    },
+    // STAGE 5: Sort by most recent comment date
+    {
+      $sort: { mostRecentCommentDate: -1 },
+    },
+    // STAGE 6: Limit results
+    {
+      $limit: movieId ? 50 : 20,
+    },
+    // STAGE 7: Shape final output
+    {
+      $project: {
+        title: 1,
+        year: 1,
+        genres: 1,
+        _id: 1,
+        imdbRating: "$imdb.rating",
+        recentComments: {
+          $map: {
+            input: "$recentComments",
+            as: "comment",
+            in: {
+              _id: "$$comment._id",
+              userName: "$$comment.name",
+              userEmail: "$$comment.email",
+              text: "$$comment.text",
+              date: "$$comment.date",
+            },
+          },
+        },
+        totalComments: { $size: "$comments" },
+      },
+    }
+  );
+
+  const results = await moviesCollection.aggregate(pipeline).toArray();
+
+  // Convert ObjectId to string for response
+  const processedResults: MovieWithCommentsResult[] = results.map((result) => ({
+    _id: result._id.toString(),
+    title: result.title,
+    year: result.year,
+    genres: result.genres,
+    imdb: result.imdbRating ? { rating: result.imdbRating } : undefined,
```

Comment:
This doesn't seem to match what the client is expecting:
```suggestion
    imdbRating: result.imdbRating,
```